tool/dumper/vkgcPipelineDumper.cpp

```diff
@@ -952,8 +952,13 @@ void PipelineDumper::updateHashForVertexInputState(const VkPipelineVertexInputSt
 // @param isRelocatableShader : TRUE if we are building relocatable shader
 void PipelineDumper::updateHashForNonFragmentState(const GraphicsPipelineBuildInfo *pipeline, bool isCacheHash,
                                                    MetroHash64 *hasher, bool isRelocatableShader) {
+  auto nggState = &pipeline->nggState;
+  bool enableNgg = nggState->enableNgg;
+
   auto iaState = &pipeline->iaState;
-  hasher->Update(iaState->topology);
+  if (enableNgg) {
```

Comment:
What about the case when you're not doing relocatable shader? Don't we then need to include it in the hash, because it affects the pal metadata?

Reply:
When I looked at it, I did not see how that happens.  The only place the field is used in Gfx9ConfigBuilder is [here](https://github.com/GPUOpen-Drivers/llpc/blob/7aa6b39b1db728a2d134d1b336cb3462fe14938c/lgc/patch/Gfx9ConfigBuilder.cpp#L1527).  The start of that function [asserts that ngg is enabled](https://github.com/GPUOpen-Drivers/llpc/blob/7aa6b39b1db728a2d134d1b336cb3462fe14938c/lgc/patch/Gfx9ConfigBuilder.cpp#L1377).

The same entry in the pal metadata get set in other places like [here](https://github.com/GPUOpen-Drivers/llpc/blob/7aa6b39b1db728a2d134d1b336cb3462fe14938c/lgc/patch/Gfx9ConfigBuilder.cpp#L1323), but the value does not come from the topology field in the pipeline state.  The geometry mode is used, which comes from the [exec model in the spir-v](https://github.com/GPUOpen-Drivers/llpc/blob/6107115c6a63e291a1b84bc175914d65efb96715/llpc/translator/lib/SPIRV/SPIRVReader.cpp#L6423).

Reply:
Ah yes, I see what you mean.